backend/templateRecs/templateRecommendation.js

```diff
@@ -0,0 +1,67 @@
+import { calculateBaseScore } from "../controllers/recommendationController.js";
+import { prisma } from "../utils/helpers.js";
+
+const calculateLocality = async (userId, creatorId) => {
+  const maxDepth = 2;
+  const visited = new Set();
+  const queue = [[userId, 0]];
+  let mutualFollowers = 0;
+
+  // Breadth first search approach to finding second degree connections
```

Comment:
Nice, so you're doing BFS from the userId w/a depth of 2, and just counting any user that matches the creator, correct?

Reply:
Yes, that's what I'm doing